tested/languages/typescript/config.py

```diff
@@ -0,0 +1,257 @@
+            "identifier": "camel_case",
+            "global_identifier": "macro_case",
+            "property": "camel_case",
+            "class": "pascal_case",
+        }
+
+    def supported_constructs(self) -> set[Construct]:
+        return {
+            Construct.OBJECTS,
+            Construct.EXCEPTIONS,
+            Construct.FUNCTION_CALLS,
+            Construct.ASSIGNMENTS,
+            Construct.HETEROGENEOUS_COLLECTIONS,
+            Construct.HETEROGENEOUS_ARGUMENTS,
+            Construct.EVALUATION,
+            Construct.DEFAULT_PARAMETERS,
+            Construct.GLOBAL_VARIABLES,
+            Construct.NAMED_ARGUMENTS,
+        }
+
+    def datatype_support(self) -> dict[AllTypes, TypeSupport]:
+        return {  # type: ignore
+            "integer": "supported",
+            "real": "supported",
+            "char": "reduced",
+            "text": "supported",
+            "string": "supported",
+            "boolean": "supported",
+            "sequence": "supported",
+            "set": "supported",
+            "map": "supported",
+            "dictionary": "supported",
+            "object": "supported",
+            "nothing": "supported",
+            "undefined": "supported",
+            "null": "supported",
+            "int8": "reduced",
+            "uint8": "reduced",
+            "int16": "reduced",
+            "uint16": "reduced",
+            "int32": "reduced",
+            "uint32": "reduced",
+            "int64": "reduced",
+            "uint64": "reduced",
+            "bigint": "supported",
+            "single_precision": "reduced",
+            "double_precision": "reduced",
+            "double_extended": "reduced",
+            "array": "reduced",
+            "list": "reduced",
+            "tuple": "reduced",
+        }
+
+    def collection_restrictions(self) -> dict[AllTypes, set[ExpressionTypes]]:
+        return {AdvancedObjectTypes.OBJECT: {BasicStringTypes.TEXT}}
+
+    def compilation(self, files: list[str], directory: Path) -> CallbackResult:
+        submission = submission_file(self)
+        main_file = list(filter(lambda x: x == submission, files))
+
+        # Create a config file to just that extends tsconfig.
+        # This way it will only run tsc on the current file.
+
+        if main_file:
+
+            config_file = {
+                "extends": str(Path(__file__).parent / "tsconfig.json"),
+                "include": [f"{main_file[0]}"],
+            }
+            with open(str(directory / "tsconfig-sub.json"), "w") as file:
+                file.write(json.dumps(config_file, indent=4))
+
+            return (
+                [
+                    "tsc",
+                    "--project",
+                    "tsconfig-sub.json",
+                ],
+                files,
+            )
+        else:
+            return [], files
+
+    def execution(self, cwd: Path, file: str, arguments: list[str]) -> Command:
+        return ["tsx", file, *arguments]
```

Comment:
This might also work without tsconfig, but even if not, it seems like we can just use a general one (as include is not mentioned [here](https://esbuild.github.io/content-types/#tsconfig-json) so it will ignored anyway.

Reply:
This works without tsconfig, so this isn't an issue.